### PR TITLE
Feature Target-SPF generic component

### PR DESCRIPTION
**Pull request by pablopareja**
imagen

### Overview

This PR adds the feature component `TargetSPF`.

### Designs

https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=1523%3A2317

### Testing instructions

Go to http://192.168.1.133:6006/?path=/story/components-features-targetspf-item--default and test the different options available.

> _⚠️NOTE: For some reason the storybook interface isn't reacting when changing boolean values in a property object, e.g. `isAllTargets: true`. See the screenshot below 👇 . Thus, in order to see the other state you might need to change the `component.stories` file_
imagen
 
### Feature relevant tickets
-

### CODE REVIEW

File: app/components/features/target-spf-item/component.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    id: 1,
+    isAllTargets: false,
+    targetValue: 0.17,
+    spfValue: 1,
+    type: Type.BIOREGIONAL,
+    surface: '30 km²',
+    name: 'Inland water',
+  },
+  onRemove: (value: TargetSPF) => console.info('Remove: ', value),
+  onChange: (value: TargetSPF) => console.info('Change', value),
```
Comment (mbarrenechea) [inline app/components/features/target-spf-item/component.stories.tsx:28]: You defined onChange only for the SPF value. 
We need to define 2 `onChangeSPF` and `onChangeTarget`

File: app/components/features/target-spf-item/types.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export enum Type {
+  BIOREGIONAL = 'bioregional',
+  SPECIES = 'species',
+  BIOREGIONAL_AND_SPECIES = 'bioregional-and-species',
+}
+
+export interface TargetSPF {
+  isAllTargets: boolean;
+  id: string;
+  targetValue: number;
```
Comment (mbarrenechea) [inline app/components/features/target-spf-item/types.ts:10]: let's simplify the name of these attributes
targetValue => target
spfValue => spf

**Review by mbarrenechea — CHANGES_REQUESTED** [review]
Good job BTW

File: app/components/features/target-spf-item/component.stories.tsx (hunk 1 above)
Comment (pablopareja) [reply app/components/features/target-spf-item/component.stories.tsx:28]: In reality the function `onChange` is triggered both when `spf` and `target` are changed - either by changing the text input value or the slider. The value passed as a parameter is an object of type `TargetSPF` having all its fields up to date.

Comment (pablopareja) [reply app/components/features/target-spf-item/component.stories.tsx:28]: @mbarrenechea ☝️

**Comment by mbarrenechea** [comment]
@pablopareja please fix the conflicts so we can merge the PR

**Review by mbarrenechea — APPROVED** [review]

**Pull request merged**